spec/cypress/integration/example-tests.spec.js

```diff
@@ -0,0 +1,119 @@
+describe("Example tests", () => {
+  beforeEach(() => {
+    cy.app("clean");
+  });
+
+  after(() => {
+    cy.app("clean");
+  });
+
+  // This is an example of a failing test - skipping because it takes a while to run
+  it.skip("should be viewable when logged out", () => {
+    cy.appFactories([["create", "course_lesson"]]);
+
+    cy.visit("/");
+
+    cy.get("#navigation").contains("Core Induction Programme").click();
+
+    cy.get("h1").should("contain", "What is the early career framework?");
+
+    cy.contains("Test Course year").click();
+
+    cy.get("h1").should("contain", "Test Course year");
+    cy.get(".govuk-govspeak").should("contain", "No content");
+
+    cy.get("a").contains("Test Course module").click();
```

Comment:
This is great as a proof of concept as long as we are mindful that this pattern will potentially create a testing architecture nightmare if not trimmed/refactored continuously - One of the more elegant test patterns is page object pattern where you abstract the things that you see on a page in different classes and as such finding what a specific page has would continuously improve.

Reply:
yep, this is all demo code